crypto/evp_extra/evp_extra_test.cc

```diff
@@ -2999,6 +2999,7 @@ TEST(EVPExtraTest, KeygenCallbacks) {
   ASSERT_EQ(EVP_PKEY_keygen_init(ctx.get()), 1);
   ASSERT_TRUE(EVP_PKEY_keygen(ctx.get(), &pkey));
   ASSERT_TRUE(pkey);
+  EVP_PKEY_free(pkey);
```

Comment:
Could this test use a unique pointer up on line 2998 to avoid needing to manually handle this?
```
bssl::UniquePtr<EVP_PKEY> pkey(EVP_PKEY_new());
```

Reply:
* B/c `EVP_PKEY_keygen` takes a `EVP_PKEY**`, our tests typically declare a local "raw" pointer and pass its address in. We typically wrap it in a smart pointer after that.
* Our static analysis indicated there was a memory leak around `pkey` in this test function. Looking more closely at [`EVP_PKEY_assign`](https://github.com/aws/aws-lc/blob/fbeb5e8452898e4f8a527447a3fa18df6d59ab3a/crypto/fipsmodule/evp/evp.c#L425) and [`EVP_PKEY_set_type`](https://github.com/aws/aws-lc/blob/fbeb5e8452898e4f8a527447a3fa18df6d59ab3a/crypto/fipsmodule/evp/evp.c#L448), I think it's just a false positive. The subsequent call to `EVP_PKEY_keygen` on the same `pkey` takes care of freeing the necessary resources prior to its reinitialization.